src/rust/multi.libgroup/template/src/template.rs

```diff
@@ -11,25 +13,77 @@ use aidoku::{
 use alloc::string::ToString;
 extern crate alloc;
 
-use crate::parser;
+use crate::{
+	helpers::{get_token, is_logged, save_token},
+	parser,
+};
 
 pub struct SocialLibSource {
 	pub site_id: &'static str,
 	pub domain: &'static str,
 	pub nsfw: &'static MangaContentRating,
-	pub cdn: &'static CDN,
 }
 
 pub struct CDN {
-	pub main: &'static str,
-	pub second: &'static str,
-	pub compress: &'static str,
+	pub main: String,
+	pub second: String,
+	pub compress: String,
 }
 
 static USER_AGENT: &str = "Mozilla/5.0 (iPhone; CPU iPhone OS 18_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1";
 static DOMAIN_API: &str = "https://api2.mangalib.me/api/";
 
 impl SocialLibSource {
+	pub fn refresh_token(&self) {
+		if is_logged() {
+			let timestamp = defaults_get("timestamp").unwrap().as_int().unwrap() / 1000;
```

Comment:
don't use `unwrap()` everywhere. you can make the function return a Result, then use `?` to handle errors.